src/Microsoft.DotNet.XHarness.Apple/Orchestration/TestOrchestrator.cs

```diff
@@ -132,10 +132,12 @@ public class TestOrchestrator : BaseOrchestrator, ITestOrchestrator
         // we didn't manage to start the app run until then.
         using var launchTimeoutCancellation = new CancellationTokenSource();
         var appRunStarted = false;
-        var task = Task.Delay(launchTimeout < timeout ? launchTimeout : timeout).ContinueWith(t =>
+        var shorterTimeout = launchTimeout < timeout ? launchTimeout : timeout;
+        var task = Task.Delay(shorterTimeout).ContinueWith(t =>

```

Comment:
There's an overload of `CancellationTokenSource` that takes `TimeSpan` after which it will be cancelled. Couldn't we use it here instead of manual delay?